astroquery/svo_fps/core.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import requests
+import io
+
+from astropy import units as u
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+from ..query import BaseQuery
+
+__all__ = ['SvoFpsClass', 'SvoFps']
+
+FLOAT_MAX = np.finfo(np.float64).max
+
+
+class SvoFpsClass(BaseQuery):
+    SVO_MAIN_URL = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+    def data_from_svo(self, query, error_msg='No data found for requested query'):
+        """Get data in response to the query send to SVO FPS
+
+        Parameters
+        ----------
+        query : dict
+            Used to create a HTTP query string i.e. send to SVO FPS to get data.
+            In dictionary, specify keys as search parameters (str) and
+            values as required. List of search parameters can be found at
+            http://svo2.cab.inta-csic.es/theory/fps/fps.php?FORMAT=metadata
+        error_msg : str, optional
+            Error message to be shown in case no table element found in the
+            responded VOTable. Use this to make error message verbose in context
+            of the query made (default is 'No data found for requested query')
+
+        Returns
+        -------
+        astropy.table.table.Table object
+            Table containing data fetched from SVO (in response to query)
+        """
+        response = self._request("GET", self.SVO_MAIN_URL, params=query)
+        response.raise_for_status()
+        votable = io.BytesIO(response.content)
+        try:
+            return parse_single_table(votable).to_table()
+        except IndexError:
+            # If no table element found in VOTable
+            raise ValueError(error_msg)
```

Comment:
maybe this can be left as an `IndexError`?

Reply:
But I think `error_msg` is also really helpful to understand why the table is not found. Perhaps, I can add custom message to same exception  (`IndexError`) by any of the approach mentioned in this [SO answer](https://stackoverflow.com/a/29442282/8252556) - let me know which one?

Reply:
you can just `raise IndexError(error_msg)`